rts-front/src/main.rs

```diff
@@ -0,0 +1,156 @@
+mod contexts;
+mod routes;
+mod screens;
+mod utils;
+
+use reqwasm::http::Request;
+use yew::prelude::*;
+use yew_router::prelude::*;
+
+use crate::contexts::*;
+use crate::routes::Route;
+use crate::screens::ai_upload::AIUpload;
+use crate::screens::login::Login;
+use crate::screens::register::Registration;
+use crate::utils::alert_message;
+
+#[function_component(HomeScreen)]
+fn home_screen() -> Html {
+    html! {
+        <h1>{ "Home screen!" }</h1>
+    }
+}
+
+#[function_component(Leaderboard)]
+fn leaderboard() -> Html {
+    html! {
+        <h1>{ "Leaderboard!" }</h1>
+    }
+}
+
+#[function_component(PageNotFound)]
+fn page_not_found() -> Html {
+    html! {
+        <h1>{ "Page not found" }</h1>
+    }
+}
+
+#[function_component(NavigationBar)]
+fn navigation_bar() -> Html {
+    let login_context = use_context::<LoginContext>().expect("no context found");
+    let history = use_history().unwrap();
+
+    let on_logout_click = {
+        let login_context = login_context.clone();
+        Callback::from(move |e: MouseEvent| {
+            // Prevent the link navigation
+            e.prevent_default();
+
+            let login_context = login_context.clone();
+            let history = history.clone();
+            // Call the backend route /api/logout
+            wasm_bindgen_futures::spawn_local(async move {
+                match Request::post("/api/logout").send().await {
+                    Ok(_) => (),
+                    Err(err) => alert_message(&format!("Couldn't log out: {}", err)),
+                }
+                // Redirect to home screen
+                history.push(Route::HomeScreen);
+                // Manually set the user as logged out (because the status updates too quickly)
+                login_context.dispatch(LoginAction::Logout)
+            });
+        })
+    };
+
+    html! {
+        <nav>
+            <ul>
+                <li><Link<Route> to={Route::HomeScreen}>{ "Home" }</Link<Route>></li>
+                <li><Link<Route> to={Route::Leaderboard}>{ "Leaderboard" }</Link<Route>></li>
+                <li><Link<Route> to={Route::AIUpload}>{ "Upload AI" }</Link<Route>></li>
+                <li><a href="https://github.com/Rolf1e/rts-m2-rust">{ "View on github" }</a></li>
+                <li>{
+                    match *login_context {
+                        LoginState::Checking => html! { "Loading…" },
+                        LoginState::LoggedOut => html! { <Link<Route> to={Route::Login}>{ "Log in" }</Link<Route>> },
+                        LoginState::LoggedIn{ username: _, user_id: _ } => html! { <a onclick={on_logout_click} href="/">{ "Log out" }</a> },
+                    }
+                }</li>
+            </ul>
+        </nav>
+    }
+}
+
+fn switch(routes: &Route) -> Html {
```

Comment:
Maybe put the switch in the `routes.rs` file. It might be less painful to find it.